lib/concurrent/atomic/read_write_lock.rb

```diff
@@ -192,7 +203,8 @@ def acquire_write_lock
     #
     # @return [Boolean] true if the lock is successfully released
     def release_write_lock
-      c = @Counter.update { |c| c-RUNNING_WRITER }
+      c = @Counter.update { |count| count-RUNNING_WRITER }
+      @WriterLock.set(nil)
```

Comment:
EDIT: When I said "another thread could conceivably acquire to lock", I meant that another thread could conceivably "pass the readers writers test once he counter." 

The flow would be:

1. Thread A  acquires writers lock
2. Thread B attempts to acquire writers locks, gets stuck waiting for Thread A
3. Thread A unsets WAITING_RUNNER
4. Thread B sets @WriterLock to Thread.current
5. Thread A sets @WriterLock to nil
6. Thread B calls acquire lock again, compares @Writer lock to Thread.current (which is nil), then spins waiting *forever* for the RUNNING_WRITER bit to be unset. Thread B can never progress.